File: mathenger-api/src/main/java/com/example/mathengerapi/models/Notification.java
package com.example.mathengerapi.models;

import com.example.mathengerapi.models.enums.NotificationType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Notification {
    @Id
    @GeneratedValue
    private Long id;
    @Enumerated(EnumType.STRING)
    @Column(length = 25, nullable = false)
    private NotificationType type;
    @ManyToOne
    @JoinColumn(name = "receiver_id", referencedColumnName = "id")
    private Account receiver;
    @ManyToOne
    @JoinColumn(name = "producer_id", referencedColumnName = "id")
    private Account producer;
    @Lob
    private String text;
    @CreationTimestamp
    private LocalDateTime time;
}

File: wpf-app/mathenger/Mathenger/Mathenger/Models/Notification.cs
using System;
using Mathenger.Models.Enums;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Mathenger.models
{
    public class Notification : BaseViewModel
    {
        [JsonProperty("id")] public long Id { get; set; }

        [JsonProperty("type")]
        [JsonConverter(typeof(StringEnumConverter))]
        public NotificationType Type { get; set; }

        [JsonProperty("receiver")] public Account Receiver { get; set; }
        [JsonProperty("producer")] public Account Producer { get; set; }
        [JsonProperty("text")] public string Text { get; set; }

        [JsonProperty("time")] public DateTime? Time { get; set; }
    }
}